src/main.ts

```diff
@@ -196,43 +252,49 @@ async function runModuleReplacements(): Promise<void> {
 
   scanSpinner.message('Scanning files');
 
-  const files = await traverseFiles(options.filesDir);
+  try {
+    const files = await traverseFiles(options.filesDir);
+    const threads = getWantedThreads(options.scanSpeed);
```

Comment:
this should probably be a concern of `scanFiles` rather than living here in the main function

once you move the thread count to a flag, you can just pass that in as a parameter here

Reply:
True, but was thinking, what if things get added and also need workers, because now the threads are calculated according the fast, medium etc

You would change to purely threads or can scan speed stay?